src/extension/settingsHelper.ts

```diff
@@ -3,88 +3,62 @@
 
 import * as Q from "q";
 import * as vscode from "vscode";
-import fs = require("fs");
 import path = require("path");
-import {ConfigurationReader} from "../common/configurationReader";
+import {ConfigurationReader, ConfigurationType} from "../common/configurationReader";
 import {Packager} from "../common/packager";
-import {FileSystem} from "../common/node/fileSystem";
 
 export class SettingsHelper {
 
-    public static get settingsJsonPath(): string {
-        return path.join(vscode.workspace.rootPath, ".vscode", "settings.json");
-    }
-
-    public static get launchJsonPath(): string {
-        return path.join(vscode.workspace.rootPath, ".vscode", "launch.json");
-    }
-
     /**
-     * Constructs a JSON object from tsconfig.json. Will create the file if needed.
+     * Path to the workspace settings file
      */
-    public static readSettingsJson(): Q.Promise<any> {
-        let settingsJsonPath: string = SettingsHelper.settingsJsonPath;
-        let fileSystem = new FileSystem();
-
-        return fileSystem.exists(settingsJsonPath)
-            .then(function(exists: boolean): Q.Promise<string> {
-                if (!exists) {
-                    return fileSystem.writeFile(settingsJsonPath, "{}")
-                        .then(() => { return "{}"; });
-                }
-
-                return fileSystem.readFile(settingsJsonPath, "utf-8");
-            })
-            .then(function(jsonContents: string): Q.Promise<any> {
-                return JSON.parse(jsonContents);
-            });
+    public static get settingsJsonPath(): string {
+        return path.join(vscode.workspace.rootPath, ".vscode", "settings.json");
     }
 
     /**
-     * Writes out a JSON configuration object to the tsconfig.json file.
+     * Returns the value for a given configuration or the default value provided.
      */
-    public static writeSettingsJson(settingsJson: any): Q.Promise<void> {
-        let settingsJsonPath: string = SettingsHelper.settingsJsonPath;
-
-        return Q.nfcall<void>(fs.writeFile, settingsJsonPath, JSON.stringify(settingsJson, null, 4));
+    public static readConfiguration(configuration: string, type: ConfigurationType, defaultValue?: any): any {
```

Comment:
I used any because it can be either string, number, boolean, array or object depending in what configuration the user is trying to read.